### PR TITLE
Add Exception and FlutterException Schema Types

### PR DESCRIPTION
**Pull request by lucaslabari**
## Goal

- As described [here](https://www.notion.so/embraceio/Exceptions-14b5111b8d01420d94457b789b22dba8), `exceptions` and `flutter exceptions` will be treated as different schema types than simple logs that don't carry any stacktrace. 
- This PR introduces these schema types but is not yet using them in prod. The hook to use this will be added in a future PR once backend is ready to receive them. 

## Testing

Added unit tests.

### CODE REVIEW

**Comment by bidetofevil** [comment]
LGTM. I think we can improve on how we define the attributes, but we can merge it like this and refactor later. 

Let me put up a PR to convert the OTel Log to show you what I mean.

**Comment by bidetofevil** [comment]
Here it is: https://github.com/embrace-io/embrace-android-sdk/pull/677

Feel free to merge your change as it is and I'll do the rebase on my end. Alternatively, you can wait until mine gets merged and you can do the rebase - whatever you feel like :)

**Review by bidetofevil — APPROVED** [review]

**Review by fractalwrench — APPROVED** [review]
LGTM

**Pull request merged**
